ios/PusherWebsocketReactNative.swift

```diff
@@ -99,17 +102,29 @@ import Foundation
         ])
         
         let key = channelName + socketID
-        authorizerMutex[key] = DispatchSemaphore(value: 0)
-        authorizerMutex[key]!.wait()
-        let authParams = authorizerResult.removeValue(forKey: key)!
-        completionHandler(PusherAuth(auth: authParams["auth"]!, channelData: authParams["channel_data"], sharedSecret: authParams["shared_secret"]))
+        let authCallback = { (authParams:[String:String]) in
+            if let authParam = authParams["auth"] {
+                completionHandler(PusherAuth(auth: authParam, channelData: authParams["channel_data"], sharedSecret: authParams["shared_secret"]))
+            } else {
+                completionHandler(PusherAuth(auth: "<missing_auth_param>:error", channelData: authParams["channel_data"], sharedSecret: authParams["shared_secret"]))
+            }
+        }
+        authorizerCompletionHandlers[key] = authCallback
+        
+        // the JS thread might not call onAuthorizer – we need to cleanup the completion handler after timeout
+        let timeout = DispatchTimeInterval.seconds(self.authorizerCompletionHandlerTimeout)
+        DispatchQueue.main.asyncAfter(deadline: .now() + timeout) {
+            if let storedAuthHandler = self.authorizerCompletionHandlers.removeValue(forKey: key) {
+                storedAuthHandler(["auth": "<authorizer_timeout>:error"])
```

Comment:
Same here – passing this just to trigger a clear error, so engineers using this SDK can debug the issues easier